framework/src/meshgenerators/StitchedMeshGenerator.C

```diff
@@ -157,7 +173,7 @@ StitchedMeshGenerator::generate()
       if (overlap_found)
       {
         const auto max_boundary_id = *base_mesh_bids.rbegin();
-        BoundaryID new_index = 0;
+        BoundaryID new_index = 1;
```

Comment:
Please add a comment about why 1 is chosen

Reply:
0 doesnt offset very well ;)

Reply:
the problem here is that we were not offsetting the index for the first one, because we were adding 0